### PR TITLE
Refactor G2P module and related process

### PR DESCRIPTION
**Pull request by lmxue**
Refactor code related to g2p, including:
* modules and utils related to g2p
* config settings
* dataset
* model input
* model inference

### CODE REVIEW

File: models/tts/valle/valle_dataset.py [hunk 1]
```diff
@@ -51,41 +43,42 @@ def __init__(self, cfg, dataset, is_valid=False):
...
             
-            ### get text tokenizer
-            text_token_path = os.path.join(
-                cfg.preprocess.processed_dir,
-                dataset,
-                cfg.preprocess.symbols_dict
-            )
-            unique_tokens = SymbolTable.from_file(text_token_path)
-            text_tokenizer = TextToken(unique_tokens.symbols, add_bos=True, add_eos=True)
+        #     ### get text tokenizer
```
Comment (zhizhengwu) [inline models/tts/valle/valle_dataset.py:51]: delete commented lines

File: models/tts/valle/valle_dataset.py [hunk 2]
```diff
@@ -218,30 +200,41 @@ def __init__(self,args, cfg):
...
+            # for utt_info in self.metadata:
+            #     dataset = utt_info["Dataset"]
+            #     uid = utt_info["Uid"]
+            #     utt = "{}_{}".format(dataset, uid)
                 
-                phone_id_seq, phn_len = text_tokenizer.get_token_id_seq(utt_info["Phone"])
-                prompt_phone_id_seq, prompt_phn_len = text_tokenizer.get_token_id_seq(utt_info["Prompt_phone"])
-                self.utt2seq[utt] = phone_id_seq
-                self.utt2pmtseq[utt] = prompt_phone_id_seq
+            #     if cfg.preprocess.phone_extractor == 'lexicon':
```
Comment (zhizhengwu) [inline models/tts/valle/valle_dataset.py:228]: too many commented lines. We need to delete them!

File: models/tts/vits/vits_trainer.py [hunk 3]
```diff
@@ -22,11 +22,26 @@
...
-    ### Following are methods only for VITS ###
+        
+        if cfg.preprocess.use_spkid and cfg.train.multi_speaker_training:
+            if cfg.model.n_speakers == 0:
+                cfg.model.n_speaker = len(self.speakers)
+            
+            
     def _build_model(self):
+        
+        # if self.cfg.preprocess.phone_extractor == 'lexicon':
```
Comment (zhizhengwu) [inline models/tts/vits/vits_trainer.py:33]: delete commented lines

**Review by RMSnow — APPROVED** [review]
approved

**Pull request merged**